control-mvt/src/main/java/org/oskari/control/mvt/GetWFSVectorTileHandler.java

```diff
@@ -61,106 +62,32 @@ public void init() {
-        }
-
-        String cacheKey = getCacheKey(layerId, srs, z, x, y);
-        byte[] cached = cache.get(cacheKey);
-        if (cached != null) {
-            params.getResponse().addHeader("Access-Control-Allow-Origin", "*");
-            if (shouldGzip(params.getRequest())) {
-                params.getResponse().addHeader("Content-Encoding", "gzip");
-                ResponseHelper.writeResponse(params, 200, MVT_CONTENT_TYPE, cached);
-            } else {
-                ByteArrayOutputStream ungzipped;
-                try {
-                    ungzipped = IOHelper.ungzip(cached);
-                } catch (IOException e) {
-                    throw new ActionException("Failed to decompress cached response", e);
-                }
-                ResponseHelper.writeResponse(params, 200, MVT_CONTENT_TYPE, ungzipped);
-            }
-            return;
-        }
-
-        // Find nearest higher resolution
-        int targetZ = grid.getZForResolution(8, -1);
-
-        List<TileCoord> wfsTiles;
-        int dz = z - targetZ;
-        if (dz < 0) {
-            int d = (int) Math.pow(2, -dz);
-            int targetX1 = x * d;
-            int targetY1 = y * d;
-            int targetX2 = (x+1) * d;
-            int targetY2 = (y+1) * d;
-            wfsTiles = new ArrayList<>(); 
-            for (int targetX = targetX1; targetX < targetX2; targetX++) {
-                for (int targetY = targetY1; targetY < targetY2; targetY++) {
-                    wfsTiles.add(new TileCoord(targetZ, targetX, targetY));
-                }
-            }
-        } else if (dz == 0) {
-            wfsTiles = Collections.singletonList(new TileCoord(z, x, y));
-        } else {
-            int div = (int) Math.pow(2, dz);
-            int targetX = x / div;
-            int targetY = y / div;
-            wfsTiles = Collections.singletonList(new TileCoord(targetZ, targetX, targetY));
-        }
-
-        SimpleFeatureCollection sfc = null;
-        for (TileCoord tile : wfsTiles) {
-            SimpleFeatureCollection tileFeatures = WFSMetaTiles.getFeatures(layer, srs, grid, tile);
-            if (tileFeatures == null) {
-                throw new ActionException("Failed to get features from service");
-            }
-            sfc = sfc == null ? tileFeatures : union(sfc, tileFeatures);
-        }
-
-        double[] bbox = grid.getTileExtent(new TileCoord(z, x, y));
-        byte[] encoded = SimpleFeaturesMVTEncoder.encodeToByteArray(sfc, layer.getName(), bbox, 4096, 256);
-
-        byte[] gzip = null;
+        final int layerId = params.getRequiredParamInt(ActionConstants.PARAM_ID);
+        final String srs = params.getRequiredParam(ActionConstants.PARAM_SRS);
+        final int z = params.getRequiredParamInt(PARAM_Z);
+        final int x = params.getRequiredParamInt(PARAM_X);
+        final int y = params.getRequiredParamInt(PARAM_Y);
+
+        final OskariLayer layer = findLayer(layerId);
```

Comment:
Add a credentials check here to know if the user has permission to view the layer?

Reply:
Will do